hack/metrics_linter/metrics_linter.go

```diff
@@ -0,0 +1,258 @@
+package main
+
+import (
+	"fmt"
+	"os"
+	"regexp"
+	"strings"
+
+	"github.com/kubevirt/hyperconverged-cluster-operator/pkg/metrics"
+)
+
+type Metric struct {
+	Name string
+	Help string
+	Type string
+}
+
+var errs []string
+
+// Do not add types to this list
+var validMetricTypes = map[string]bool{
+	"counter":   true,
+	"gauge":     true,
+	"histogram": true,
+	"summary":   true,
+}
+
+var operator = os.Args[1]
+var subOperator = os.Args[2]
+
+// lists of metrics to be excluded in some checks
+var hyperconvergedMetrics = []string{
+	"kubevirt_hyperconverged_operator_health_status",
+}
+
+var StatusMetrics = []string{
+	"kubevirt_hyperconverged_operator_health_status",
+	"kubevirt_hco_system_health_status",
+	"kubevirt_hco_node_status_max_images",
+}
+
+// Units and their possible prefixes recognized by this library. More can be
+// added over time as needed
+var (
+	// map a unit to the appropriate base unit.
+	units = map[string]string{
+		// Base units
+		"amperes": "amperes",
+		"bytes":   "bytes",
+		"celsius": "celsius", // Also allow Celsius because it is common in typical Prometheus use cases
+		"grams":   "grams",
+		"joules":  "joules",
+		"kelvin":  "kelvin", // SI base unit, used in special cases (e.g. color temperature, scientific measurements)
+		"meters":  "meters", // Both American and international spelling permitted
+		"metres":  "metres",
+		"seconds": "seconds",
+		"volts":   "volts",
+
+		// Non base units
+		// Time
+		"minutes": "seconds",
+		"hours":   "seconds",
+		"days":    "seconds",
+		"weeks":   "seconds",
+		// Temperature
+		"kelvins":    "kelvin",
+		"fahrenheit": "celsius",
+		"rankine":    "celsius",
+		// Length
+		"inches": "meters",
+		"yards":  "meters",
+		"miles":  "meters",
+		// Bytes
+		"bits": "bytes",
+		// Energy
+		"calories": "joules",
+		// Mass
+		"pounds": "grams",
+		"ounces": "grams",
```

Comment:
do we need these?

Reply:
@nunnatsa I tried to import it from here [promlint] (https://github.com/prometheus/client_golang/tree/v1.14.0/prometheus/testutil/promlint) but it's not working with just import.